From 45738a5157cee9ccc1754fd6a38352505b9ece6f Mon Sep 17 00:00:00 2001
From: Ewan Milne <emilne@redhat.com>
Date: Wed, 20 Aug 2014 18:59:31 -0400
Subject: [scsi] scsi_debug: fix duplicate dif_errors increment

Message-id: <1408561176-18133-53-git-send-email-emilne@redhat.com>
Patchwork-id: 88457
O-Subject: [RHEL7.1 PATCH e-stor 52/57] [SCSI] scsi_debug: fix duplicate dif_errors increment
Bugzilla: 1132100
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Ewan D. Milne" <emilne@redhat.com>

upstream commit c613d061ce01be1f4e195f4666d4a99e5284cb5d

Tested by Ewan D. Milne and Barry Donahue with a combination
of manual and automated testing.  No issues seen.

upstream commit c613d061ce01be1f4e195f4666d4a99e5284cb5d
Author: Akinobu Mita <akinobu.mita@gmail.com>
Date:   Wed Feb 26 22:57:00 2014 +0900

    [SCSI] scsi_debug: fix duplicate dif_errors increment

    It is unnecessary to increase dif_errors in dif_verify(), because the
    caller will increment it when dif_verify() detects failure.

    This bug was introduced by commit beb40ea42bd6 ("[SCSI] scsi_debug:
    reduce duplication between prot_verify_read and prot_verify_write")

    Signed-off-by: Akinobu Mita <akinobu.mita@gmail.com>
    Acked-by: Martin K. Petersen <martin.petersen@oracle.com>
    Reviewed-by: Douglas Gilbert <dgilbert@interlog.com>
    Signed-off-by: James Bottomley <JBottomley@Parallels.com>

Signed-off-by: Ewan D. Milne <emilne@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/scsi_debug.c b/drivers/scsi/scsi_debug.c
index 9cd211e..1a42880 100644
--- a/drivers/scsi/scsi_debug.c
+++ b/drivers/scsi/scsi_debug.c
@@ -1780,7 +1780,6 @@ static int dif_verify(struct sd_dif_tuple *sdt, const void *data,
      be32_to_cpu(sdt->ref_tag) != ei_lba) {
   pr_err("%s: REF check failed on sector %lu\n",
    __func__, (unsigned long)sector);
-   dif_errors++;
   return 0x03;
  }
  return 0;
-- 
1.7.1